dpgen/generator/run.py

```diff
@@ -3165,20 +3165,26 @@ def run_fp (iter_index,
 def post_fp_check_fail(iter_index,
                        jdata,
                        rfailed = None) :
+
     ratio_failed =  rfailed if rfailed else jdata.get('ratio_failed',0.05)
     iter_name = make_iter_name(iter_index)
     work_path = os.path.join(iter_name, fp_name)
     fp_tasks = glob.glob(os.path.join(work_path, 'task.*'))
     fp_tasks.sort()
     if len(fp_tasks) == 0 :
         return
-    # check fail according to tag_failure
-    fp_failed_tags = glob.glob(os.path.join(work_path, 'task.*', 'tag_failure*'))
-    fp_failed_tasks = [os.path.dirname(ii) for ii in fp_failed_tags]
-    fp_failed_tasks = list(set(fp_failed_tasks))
-
     ntask = len(fp_tasks)
-    nfail = len(fp_failed_tasks)
+    nfail = 0
+
+    # check fail according to the number of collected data
+    sys_data = glob.glob(os.path.join(work_path, "data.*"))
+    sys_data.sort()
+    nframe = 0
+    for ii in sys_data :
+        sys = dpdata.LabeledSystem(ii, fmt = 'deepmd/raw')
```

Comment:
```suggestion
        sys = dpdata.LabeledSystem(ii, fmt = 'deepmd/npy')
```

It looks like some styles don't output raw files.

Reply:
Thanks for suggestion. Still, I am wondering how to fix unit tests. Orz